### PR TITLE
Improve summarize performance by using mapslices

### LINKED ISSUE #788: Improve loc_trajectory() function performance
Currently the `ADRIA.loc_trajectory()` function takes a long amount of time to run, particularly for GBR wide analyses. 
Possible speed up method is to use `mapslices()` function and specify the dimension to be reduced (`:scenarios`).

### PR DESCRIPTION
**Pull request by Zapiano**
The previous approach, using JuliannedArrays.Slices, is taking too much to run for some unknown reason. Using `mapslices` seems to fix the problem.

closes #788

### CODE REVIEW

**Review by BG-AIMS — COMMENTED** [review]
I think these are good changes.

**Comment by Zapiano** [comment]
@ConnectedSystems @BG-AIMS I've changed the approach here, going back to JuliennedArrays, and now it's much faster than with `mapslices` (as mentioned here: https://github.com/open-AIMS/ADRIA.jl/issues/788#issuecomment-2373154482 )

**Comment by Zapiano** [comment]
Just to give an idea:

Using `mapslices`:
image

Using JuliennedArrays with `read()`:
image

**Comment by Zapiano** [comment]
@ConnectedSystems I still have to test this, but I tried dynamically determining if there's enough space to `read(data)`. Let me know if that's what you were thinking.

**Comment by Zapiano** [comment]
Quick comment: I changed the threshold to be 70% of available space in RAM, to have a safety margin to account for memory spikes and other unpredicted memory usages that could cause a problem.

File: src/metrics/site_level.jl [hunk 1]
```diff
@@ -86,12 +86,26 @@ YAXArray with summary metric for the remaining axis.
...
+    # available space in RAM
+    data_size = Base.summarysize(data)
+    free_ram = Sys.free_memory()
+    proportional_usage = data_size / free_ram
+
+    # Only use this approach when data occupies more than 70% of available space in RAM
+    if proportional_usage > 0.7 * free_ram
+        # `D.` is ensuring the returned YAXArray has the same type as the input `data`
+        return D.(mapslices(metric, data; dims=alongs_axis))
+    end
```
Comment (ConnectedSystems) [inline src/metrics/site_level.jl:101]: Suggesting a more readable if statement

```suggestion
    data_size = Base.summarysize(data)
    
    # Only use this approach when data occupies more than 70% of available space in RAM
    available_threshold = Sys.free_memory() * 0.7
    proportional_usage = data_size / free_ram
    if proportional_usage > available_threshold 
        # `D.` is ensuring the returned YAXArray has the same type as the input `data`
        return D.(mapslices(metric, data; dims=alongs_axis))
    end
```

File: src/metrics/site_level.jl [hunk 2]
```diff
@@ -86,18 +86,25 @@ YAXArray with summary metric for the remaining axis.
...
+    # The approach using JuliennedArrays is faster then directly passing the YAXArray to
+    # mapslices only when we `read(data)`. Since `read(data)` loads all the data from disk
+    # into memory, we only want to use that approach when there is a reasonable amount of
+    # available space in RAM
     data_size = Base.summarysize(data)
     free_ram = Sys.free_memory()
-    if data_size > free_ram
+    proportional_usage = data_size / free_ram
+
+    # Only use this approach when data occupies more than 70% of available space in RAM
```
Comment (ConnectedSystems) [inline src/metrics/site_level.jl:97]:
```suggestion
    # Only use this approach when data occupies more than 70% of available RAM
```

For conciseness 😊

**Review by ConnectedSystems — APPROVED** [review]
Thanks @Zapiano

**Pull request merged**